backend/audit/views.py

```diff
@@ -428,6 +430,61 @@ def get(self, request, *args, **kwargs):
             raise PermissionDenied("You do not have access to this audit.")
 
 
+class AuditInfoFormView(SingleAuditChecklistAccessRequiredMixin, generic.View):
+    def get(self, request, *args, **kwargs):
+        report_id = kwargs["report_id"]
+        try:
+            sac = SingleAuditChecklist.objects.get(report_id=report_id)
+
+            context = {
+                "auditee_name": sac.auditee_name,
+                "report_id": report_id,
+                "auditee_uei": sac.auditee_uei,
+                "user_provided_organization_type": sac.user_provided_organization_type,
+                "agency_names": get_agency_names(),
+            }
+
+            # TODO: check if there's already a form in the DB and let the user know
+            # context['already_submitted'] = ...
+
+            return render(request, "audit/audit-info-form.html", context)
+        except SingleAuditChecklist.DoesNotExist:
+            raise PermissionDenied("You do not have access to this audit.")
+        except Exception as e:
+            logger.info("Enexpected error in AuditInfoFormView get.\n", e)
+            raise BadRequest()
+
+    def post(self, request, *args, **kwargs):
+        report_id = kwargs["report_id"]
+
+        try:
+            sac = SingleAuditChecklist.objects.get(report_id=report_id)
+            form = AuditInfoForm(request.POST, request.FILES)
+
+            if form.is_valid():
+                # Save the form, yay!
+                print("Saving audit info form!")
+                return redirect(reverse("audit:SubmissionProgress", args=[report_id]))
+            else:
+                form.clean_booleans()
+                print("hello", form.cleaned_data)
```

Comment:
Does `views.py` have `logging` imported? If so, please feel free to include some logging statements. Or, perhaps we should remove the `print` statements.

Point being: having some noise come from the submission process is fine. Having it go to the logger means it will (ultimately) make its way to our New Relic instance. This is fine, too. (I do want to do a walk of the code and add logging throughout, so we can debug, etc. But, this is a *should* for go-live, not a *must*, as crash logs will still make it through.)

Either way, don't be afraid to include some logging. Just make sure the logging message makes clear what you're doing (instead of just `"hello"`, for example, from a few lines down.)

Reply:
Yep, it sure does. The prints are from me neglecting to clean up after myself. The print from the audit info form has been changed to use logger, and the others have been removed. Thanks for catching that!